### ISSUE TITLE
multiple `binary` with :target may occur error

### ISSUE THREAD

**Issue by delphinus35** [caskroom/homebrew-cask#9227]
I'm writing a cask that has multiple `binary` artifacts.
https://github.com/delphinus35/homebrew-versions/blob/macvim-kaoriya-improved-cask/Casks/macvim-kaoriya.rb#L23

```ruby
executables.each { |e| binary mvim, :target => e }
```

But this cask will fail in installation with this error.

```
$ brew cask uninstall macvim-kaoriya
...

Error: comparison of Array with Array failed

  Most likely, this means you have an outdated version of homebrew-cask. Please run:

      brew update && brew upgrade brew-cask && brew cleanup && brew cask cleanup

  If this doesn’t fix the problem, please report this bug:

      https://github.com/caskroom/homebrew-cask/issues

/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/artifact/symlinked.rb:93:in `sort'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/artifact/symlinked.rb:93:in `uninstall_phase'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/installer.rb:272:in `block in uninstall_artifacts'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/installer.rb:270:in `each'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/installer.rb:270:in `uninstall_artifacts'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/installer.rb:261:in `uninstall'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/cli/uninstall.rb:10:in `block in run'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/cli/uninstall.rb:6:in `each'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/cli/uninstall.rb:6:in `run'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/cli.rb:78:in `run_command'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/hbc/cli.rb:118:in `process'
/usr/local/Cellar/brew-cask/0.53.0/rubylib/brew-cask-cmd.rb:18:in `<main>'
```

[symlinked.rb][]#L92 call `sort` with Arrays in Array. I remove this `sort`, then the cask finished successfully. Why this `sort` is here? Is it still needed?

[symlinked.rb]: https://github.com/caskroom/homebrew-cask/blob/master/lib/hbc/artifact/symlinked.rb#L92

**Comment by vitorgalvao** [comment]
Please post your cask as we have it, so we can better diagnose this.

**Comment by delphinus35** [comment]
The link of my cask is above.
https://github.com/delphinus35/homebrew-versions/blob/macvim-kaoriya-improved-cask/Casks/macvim-kaoriya.rb#L23

**Comment by vitorgalvao** [comment]
Ah, apologies, I misread and jumped the gun.

As for your question as to why the `sort` is there, it’s [explained in the comment](https://github.com/caskroom/homebrew-cask/blob/b79d4cd3871ee393cd80c9de7f5bcf22ced36d56/lib/hbc/artifact/symlinked.rb#L91). As for it still being needed, we dropped support for old ruby versions a while back, so perhaps it isn’t needed anymore.

Pinging  @rolandwalker and @ndr-qef, on that one.

**Comment by ndr-qef** [comment]
The error suggests that we are trying to sort arrays containing non-comparable values.

However, before considering that, we should address a potential misunderstanding: `:target` does not concern the source binary, it merely sets the symlink name. What the [line you quoted](https://github.com/delphinus35/homebrew-versions/blob/macvim-kaoriya-improved-cask/Casks/macvim-kaoriya.rb#L23) would do is create multiple symlinks of the `mvim` binary. Is that your intention?

**Comment by delphinus35** [comment]
> create multiple symlinks of the mvim binary.
Yes. That's my cask's goal.

I wrote this with referring to [homebrew's macvim formula][macvim]. It makes many mvim => xxxx symlinks with the same way. Perhaps is this an unusual way?

[macvim]: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/macvim.rb#L99-102

**Comment by rolandwalker** [comment]
Hi!

As the code comment suggests, the `sort` was a compatibility hack from when we supported multiple Ruby versions.  And the hack was incomplete, as this bug reveals.

Now that we only support Ruby 2.0+, it is safe to remove the `sort`, if the tests pass.  There are several other places we should remove similar `sort`s, both in code and in the test suite.

**Comment by vitorgalvao** [comment]
Fixed (finally) in #11110.
